pkg/backend/httpstate/backend.go

```diff
@@ -1234,6 +1260,51 @@ func (b *cloudBackend) PromptAI(
 	return res, nil
 }
 
+func (b *cloudBackend) ShowCopilotErrorSummary(
+	ctx context.Context, pulumiOutput []string, stackRef backend.StackReference, opts display.Options,
+) {
+	summary, err := b.DoSummarizeErrorWithCopilot(ctx, pulumiOutput, stackRef, opts)
+	fmt.Println(opts.Color.Colorize(colors.SpecHeadline + "Summary:" + colors.Reset))
+	if err != nil {
+		fmt.Printf("error summarizing update output: %s", err)
+	}
+	for _, line := range summary {
+		fmt.Println("  " + opts.Color.Colorize(colors.BrightGreen+line+colors.Reset))
+	}
+	// nice empty line
+	fmt.Println()
+}
+
+func (b *cloudBackend) DoSummarizeErrorWithCopilot(
+	ctx context.Context, pulumiOutput []string, stackRef backend.StackReference, opts display.Options,
+) ([]string, error) {
+	if len(pulumiOutput) == 0 {
+		return nil, nil
+	}
+
+	stackID, err := b.getCloudStackIdentifier(stackRef)
+	if err != nil {
+		return nil, err
+	}
+	orgName := stackID.Owner
+
+	model := opts.CopilotSummaryModel
+	maxSummaryLen := opts.CopilotSummaryMaxLen
+
+	startTime := time.Now()
+	summary, err := b.client.SummarizeErrorWithCopilot(ctx, orgName, pulumiOutput, model, maxSummaryLen)
+	if err != nil {
+		return nil, err
+	}
+
+	elapsedMs := time.Since(startTime).Milliseconds()
+
+	summaryHeader := fmt.Sprintf("✨ AI-generated summary (%d ms):", elapsedMs)
```

Comment:
We only support emojis on certain platforms.  We should keep following that pattern (https://github.com/pulumi/pulumi/blob/master/sdk/go/common/util/cmdutil/console.go#L35)

Reply:
TIL `--emoji` neat.